### PR TITLE
backends/eopkg: Clean up some of the memory handling

### PR DESCRIPTION
**Pull request by EbonJaeger**
Cleans up the memory handling around getting the backend filename.

### CODE REVIEW

File: backends/eopkg/pk-backend-eopkg.c [hunk 1]
```diff
@@ -25,26 +25,32 @@
...
 
-void
-eopkg_get_backend_filename(char *backend_filename)
+static const gchar
+*eopkg_get_backend_filename ()
 {
-    // char backend_filename = malloc(sizeof(char));
-    if (access("/usr/share/PackageKit/helpers/eopkg/eopkgBackend.bin", F_OK) == 0) {
-        strcpy(backend_filename, "eopkgBackend.bin");
+    g_autoptr (GFile) backend_file;
```
Comment (ikeycode) [inline backends/eopkg/pk-backend-eopkg.c:33]: Please default initialiaze as `NULL`

File: backends/eopkg/pk-backend-eopkg.c [hunk 2]
```diff
@@ -25,26 +25,32 @@
...
 {
-    char backend_filename[30];
-    eopkg_get_backend_filename(backend_filename);
+    const gchar *backend_filename = NULL;
+
+    backend_filename = eopkg_get_backend_filename ();
+
     pk_backend_spawn_helper (spawn, job, backend_filename, "repair_system", NULL);
     pk_backend_job_finished (job);
+    g_free ((gchar *) backend_filename);
```
Comment (ikeycode) [inline backends/eopkg/pk-backend-eopkg.c:53]: Why are we doing `g_free` on statically allocated memory? :)

**Review by ikeycode — CHANGES_REQUESTED** [review]
So if we're returning `const char*` (aka `const gchar*`) from statically initialised memory, remember the compiler is actually inserting a NUL terminated `const char[LEN]` for us, and this should *not* be freed by anyone.

Also remember to initialise pretty much anything =)

**Comment by EbonJaeger** [comment]
@ikeycode noted, thanks! Does this look more correct?

**Comment by ikeycode** [comment]
It does! I wonder if `g_file_test` might be less allocation heavy now? Using `G_FILE_TEST_EXISTS`, as I think it just maps to `access(somePath, X_OK)` ?

**Comment by EbonJaeger** [comment]
It looks like `g_file_query_exists` allocates a `GFileInfo`, and if it isn't `NULL`, the file exists. `g_file_test` with that option does indeed just use the `access` syscall. The docs make a note with `g_file_test` when it uses that syscall that I'm not sure if it would matter for our use or not:

> [...] if your program is setuid or setgid it means that these tests will give you the answer for the real user ID and group ID, rather than the effective user ID and group ID.

**Comment by ikeycode** [comment]
Should be good, the `packagekitd` binary is usually `0755` and run *as* root, but not `4755` afaik

**Review by ikeycode — APPROVED** [review]

**Pull request merged**